main/src/library/Foldable.flix

```diff
@@ -0,0 +1,165 @@
+
+    ///
+    /// Returns the result of applying `combine` to all the elements `t`,
+    /// using `empty` as the initial value.
+    ///
+    pub def combineAll[t :# Type -> Type : Foldable, a : Monoid](t: t[a]): a =
+        Foldable.foldLeft((acc, x) -> Monoid.combine(acc, x), Monoid.empty(), t)
+
+    ///
+    /// Returns the number of elements in `t` that satisfy the predicate `f`.
+    ///
+    pub def count[t :# Type -> Type : Foldable, a](f: a -> Bool, t: t[a]): Int32 =
+        Foldable.foldLeft((acc, x) -> if (f(x)) acc + 1 else acc, 0, t)
+
+    ///
+    /// Returns `true` if and only if at least one element in `t` satisfies the predicate `f`.
+    ///
+    /// Returns `false` if `t` is empty.
+    ///
+    pub def exists[t :# Type -> Type : Foldable, a](f: a -> Bool, t: t[a]): Bool =
+        Foldable.foldLeft((acc, x) -> acc or f(x), false, t)
+
+    ///
+    /// Returns `true` if and only if all elements in `t` satisfy the predicate `f`.
+    ///
+    /// Returns `true` if `t` is empty.
+    ///
+    pub def forall[t :# Type -> Type : Foldable, a](f: a -> Bool, t: t[a]): Bool =
+        Foldable.foldLeft((acc, x) -> acc and f(x), true, t)
+
+    ///
+    /// Optionally returns the first element of `t` that satisfies the predicate `f` when searching from left to right.
+    ///
+    /// Returns `None` if `t` is empty.
+    ///
+    pub def findLeft[t :# Type -> Type : Foldable, a](f: a -> Bool, t: t[a]): Option[a] =
+        Foldable.foldLeft((acc, x) -> Option.withDefault(acc, if (f(x)) Some(x) else None), None, t)
+
+    ///
+    /// Optionally returns the first element of `t` that satisfies the predicate `f` when searching from right to left.
+    ///
+    /// Returns `None` if `t` is empty.
+    ///
+    pub def findRight[t :# Type -> Type : Foldable, a](f: a -> Bool, t: t[a]): Option[a] =
+        Foldable.foldRight((x, acc) -> Option.withDefault(acc, if (f(x)) Some(x) else None), None, t)
+
+    ///
+    /// Returns the result of mapping each element and combining the results.
+    ///
+    pub def foldMap[t :# Type -> Type : Foldable, a, b : Monoid](f: a -> b, t: t[a]): b =
+        Foldable.foldLeft((acc, x) -> Monoid.combine(f(x), acc), Monoid.empty(), t)
+
+    ///
+    /// Returns true if and only if `t` is empty.
+    ///
+    pub def isEmpty[t :# Type -> Type : Foldable, a](t: t[a]): Bool =
+        Foldable.foldLeft((_, _) -> false, true, t)
+
+    ///
+    /// Returns `true` if and only if the element `x` is in `t`.
+    ///
+    pub def memberOf[t :# Type -> Type : Foldable, a: Eq](x: a, t: t[a]): Bool =
+        Foldable.foldLeft((acc, y) -> acc or x == y, false, t)
+
+    ///
+    /// Optionally finds the smallest element of `t` according to the `Order` on `a`.
+    ///
+    /// Returns `None` if `t` is empty.
+    ///
+    pub def minimum[t :# Type -> Type : Foldable, a: Order](t: t[a]): Option[a] =
+        Foldable.foldLeft((acc, x) -> Some(Option.foldLeft((z, v) -> Order.min(z, v), x, acc)), None, t)
```

Comment:
Here you can use `reduceLeft` right? :D

Same for the other minimum/maximum functions.

Reply:
Indeed! I implemented them before I saw you had added `reduce` to the list. I refactored it now.